### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by gigamesh**

### CODE REVIEW

File: logos/generic-op.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...fs><circle class="cls-1" cx="59.22" cy="59.22" r="59.22"/><circle class="cls-2" cx="59.35" cy="59.11" r="48.13"/><path class="cls-3" d="M38.92,78.33A16.32,16.32,0,0,1,29,75.48a9.77,9.77,0,0,1-3.81-8.25,18.21,18.21,0,0,1,.25-2.75q.66-3.66,1.88-8.81,3.47-14,17.88-14a17.79,17.79,0,0,1,7,1.32,10.72,10.72,0,0,1,4.89,3.87A10.39,10.39,0,0,1,58.88,53a17.84,17.84,0,0,1-.25,2.7q-.77,4.53-1.84,8.81-1.77,7-6.16,10.44T38.92,78.33ZM39.64,71a7.21,7.21,0,0,0,4.83-1.68,9.34,9.34,0,0,0,2.9-5.14c.79-3.2,1.38-6,1.79-8.36a12.11,12.11,0,0,0,.2-2.18q0-4.64-4.84-4.64a7.4,7.4,0,0,0-4.88,1.68,9.55,9.55,0,0,0-2.86,5.14c-.61,2.28-1.22,5.06-1.83,8.36a11.24,11.24,0,0,0-.2,2.13Q34.75,71,39.64,71Z" transform="translate(-0.5 -0.5)"/><path class="cls-3" d="M61.22,77.82a1.1,1.1,0,0,1-.87-.36,1.19,1.19,0,0,1-.15-.91l7-33.1a1.42,1.42,0,0,1,.56-.91,1.54,1.54,0,0,1,1-.36H82.3q5.65,0,9.06,2.34a7.67,7.67,0,0,1,3.46,6.77,12.45,12.45,0,0,1-.3,2.65q-1.27,5.85-5.14,8.66T78.89,65.4H72L69.67,76.55a1.42,1.42,0,0,1-.56.91,1.54,1.54,0,0,1-1,.36Zm18-19.45A6.06,6.06,0,0,0,83,57.2a5.48,5.48,0,0,0,2.14-3.36,9.14,9.14,0,0,0,.15-1.53,2.86,2.86,0,0,0-.86-2.24,4.3,4.3,0,0,0-3-.81H75.32l-1.93,9.11Z" transform="translate(-0.5 -0.5)"/></svg>
```
Comment (gigamesh) [inline logos/generic-op.svg:1]: this should have been merged earlier

**Comment by snario** [comment]
What is this value supposed to be, where is it used?

**Comment by gigamesh** [comment]
> What is this value supposed to be, where is it used?


I _think_ this should only result in an error if a user tries to withdraw, because they'd be attempting to withdraw from an address without a balance.

```
l2Receipt = await contracts.l2.bridge.connect(signer).withdraw(contracts.l2.token.address, amount, 0, '0x');
```

File: src/validate.ts [hunk 2]
```diff
@@ -23,42 +47,101 @@ const networkMap = {
...
+          `Contract decimals mismatch. ${decimals} !== ${tokenData.decimals} \nAddress: ${tokenData.address}`
         );
       }
+
+      if (tokenData.extensions?.optimismBridgeAddress) {
+        try {
+          await validateBridgeAddress(tokenData);
+        } catch (err) {
+          throw Error(err.message);
+        }
```
Comment (snario) [inline src/validate.ts:85]: This seems redundant

**Review by snario — APPROVED** [review]

**Pull request merged**